cpp/src/io/orc/writer_impl.hpp

```diff
@@ -226,6 +228,7 @@ class writer::impl {
    * @return The statistic blobs
    */
   std::vector<std::vector<uint8_t>> gather_statistic_blobs(
+    const table_device_view& table,
     orc_column_view const* columns,
```

Comment:
We could create an `orc_table` derived from table and store all of the orc information in that class. As of now, we do need to pass the `table` so that `create_leaf_column_device_views` can be called within this function.